egret/models/acopf.py

```diff
@@ -228,18 +228,31 @@ def _create_base_power_ac_model(model_data, include_feasibility_slack=False):
                                                       branches=branches
                                                       )
 
-    ### declare the generator cost objective
+    # declare the generator cost objective
+    pw_cost_gens = list()
+    p_cost = gen_attrs['p_cost']
+    if 'q_cost' in gen_attrs:
+        q_cost = gen_attrs['q_cost']
+    else:
+        q_cost = dict()
+    for gen_name in gen_attrs['names']:
+        if p_cost[gen_name]['cost_curve_type'] == 'piecewise' or (gen_name in q_cost and q_cost[gen_name]['cost_curve_type'] == 'piecewise'):
+            pw_cost_gens.append(gen_name)
+
+    if len(pw_cost_gens) > 0:
+        libgen.declare_var_pg_cost(model=model, index_set=pw_cost_gens)
+        libgen.declare_var_qg_cost(model=model, index_set=pw_cost_gens)
+        libgen.declare_piecewise_cost_cons(model=model, index_set=pw_cost_gens, p_costs=p_cost, q_costs=q_cost)
+
```

Comment:
It would be nice if it were easy to insert this logic into the other OPF models.